### PR TITLE
fix(Field.Upload): prevent `fileHandler` to run when it is validation errors 

### PR DESCRIPTION
**Pull request by langz**
fixes https://github.com/dnbexperience/eufemia/issues/4495

[CSB](https://codesandbox.io/p/sandbox/fileupload-validation-on-async-upload-forked-r2r4kd?workspaceId=ws_LbiDwJdEBdbwxH2AkyAyVE)

### CODE REVIEW

File: packages/dnb-eufemia/src/extensions/forms/Field/Upload/Upload.tsx [hunk 1]
```diff
@@ -145,41 +138,60 @@ function UploadComponent(props: Props) {
...
+            ...incomingFile,
+            isLoading: false,
+          }
+          const foundIndex = newFilesLoading.findIndex(
+            (newFile) => newFile.isLoading
+          )
+          if (foundIndex >= 0) {
+            newFilesLoading[foundIndex] = incomingFileObj
+          } else {
+            // if there's more files returned than there's loading (edge case), add them to end of array.
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Field/Upload/Upload.tsx:174]:
```suggestion
            // if there's more files incoing than there's loading (edge case), add them to end of array.
```

**Review by tujoworker — APPROVED** [review]
Nice 🌟

**Comment by langz** [comment]
I'll look into the failing tests 👀

File: packages/dnb-eufemia/src/extensions/forms/Field/Upload/Upload.tsx [hunk 2]
```diff
@@ -172,14 +179,14 @@ function UploadComponent(props: Props) {
 
         const updatedFiles = [
           ...filesRef.current.slice(0, indexOfFirstNewFile),
-          ...uploadedFiles,
+          ...newFilesLoading,
           ...filesRef.current.slice(indexOfFirstNewFile + newFiles.length),
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Field/Upload/Upload.tsx:183]: Todo look into newFiles.length vs newFilesLoading
